classes/module/Module.php

```diff
@@ -983,8 +983,9 @@ public function disable($force_all = false)
             $result &= $this->uninstallOverrides();
         }
 
-        // Disable module for all shops
-        $result &= Db::getInstance()->delete('module_shop', '`id_module` = ' . (int) $this->id);
+        // Disable module for all shops or contextual shops
+        $whereIdShop = $force_all ? '' : ' AND `id_shop` IN(' . implode(', ', Shop::getContextListShopID()) . ')';
```

Comment:
can you please check if multi-shop feature is enabled ?

Reply:
This is a legacy function and we never made this check before, I thought about it and realized that it's actually not necessary.

Except if you see a use case where it makes sense ?